spot_rl_experiments/spot_rl/ros_img_vis.py

```diff
@@ -165,37 +168,87 @@ class SpotRosVisualizer(VisualizerMixin, SpotRobotSubscriberMixin):
     no_raw = False
     proprioception = False
 
+    # Define a timeout duration (in seconds)
+    TIMEOUT_DURATION = 5
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.last_seen = {topic: time.time() for topic in self.msgs.keys()}
         self.fps = {topic: deque(maxlen=10) for topic in self.msgs.keys()}
 
+    def is_empty_image(self, img):
+        """Determine if an image is empty or has no meaningful data"""
+        if np.all(img == 0):
+            print("Image is all zeros")
+            return True
+        return False
+
     def generate_composite(self):
         if not any(self.updated.values()):
             # No imgs were refreshed. Skip.
             return None
 
         refreshed_topics = [k for k, v in self.updated.items() if v]
+        print("Available topics in self.msgs:", self.last_seen)
 
         # Gather latest images
         raw_msgs = [self.msgs[i] for i in RAW_IMG_TOPICS]
         processed_msgs = [self.msgs[i] for i in PROCESSED_IMG_TOPICS]
 
         raw_imgs = [self.msg_to_cv2(i) for i in raw_msgs if i is not None]
-
-        # Replace any Nones with black images if raw version exists. We (safely) assume
-        # here that there is no processed image anyway if the raw image does not exist.
         processed_imgs = []
+        # Handle processed messages and fill with zeros if needed
         for idx, raw_msg in enumerate(raw_msgs):
             if processed_msgs[idx] is not None:
                 processed_imgs.append(self.msg_to_cv2(processed_msgs[idx]))
-            elif processed_msgs[idx] is None and raw_msg is not None:
+            else:
                 processed_imgs.append(np.zeros_like(raw_imgs[idx]))
 
-        # Crop gripper images
+        # Crop and process images as needed
         if raw_msgs[1] is not None:
             for imgs in [raw_imgs, processed_imgs]:
                 imgs[1] = imgs[1][:, 124:-60]
+        raw_imgs[1] = cv2.convertScaleAbs(raw_imgs[1], alpha=0.03)
+        processed_imgs[2] = cv2.resize(processed_imgs[2], (640, 480))
+        processed_imgs[3] = cv2.convertScaleAbs(processed_imgs[3], alpha=0.03)
+        for topic in RAW_IMG_TOPICS + PROCESSED_IMG_TOPICS:
```

Comment:
Lets separate this into two for loops instead of having one for loop and using if statements inside. 

```
for topic in RAW_IMG_TOPICS:
    idx = RAW_IMG_TOPICS.index(topic)
    < continue code >
for topic in PROCESSED_IMG_TOPICS:
   idx = PROCESSED_IMG_TOPICS.index(topic)
    < continue code >
```